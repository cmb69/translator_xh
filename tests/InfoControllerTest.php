<?php

namespace Translator;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeSystemChecker;
use Plib\SystemChecker;
use Plib\View;
use Translator\Model\Service;

class InfoControllerTest extends TestCase
{
    private Service $service;
    private SystemChecker $systemChecker;
    private View $view;

    protected function setUp(): void
    {
        $this->service = new Service("", "", "");
        $this->systemChecker = new FakeSystemChecker();
        $this->view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["translator"]);
    }

    private function sut(): InfoController
    {
        return new InfoController(
            "./cmsimple/languages/",
            "./plugins/",
            $this->service,
            $this->systemChecker,
            $this->view
        );
    }

    public function testRendersPluginInfo(): void
    {
        $response = $this->sut()();
        $this->assertSame("Translator 1.0beta8", $response->title());
        Approvals::verifyHtml($response->output());
    }
}
