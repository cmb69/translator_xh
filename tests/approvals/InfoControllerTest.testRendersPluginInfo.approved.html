
<article class="translator_pluginfo">
  <h1>Translator 1.1-dev</h1>
  <section class="translator_syscheck">
    <h2>System check</h2>
    <p class="xh_fail">Checking that the PHP version is at least 7.4.0 … failure</p>
    <p class="xh_fail">Checking that the PHP extension 'zlib' is loaded … failure</p>
    <p class="xh_fail">Checking that the CMSimple_XH version is at least 1.7.0 … failure</p>
    <p class="xh_fail">Checking that the Plib_XH version is at least 1.10 … failure</p>
    <p class="xh_warning">Checking that './cmsimple/languages/' is writable … warning</p>
    <p class="xh_warning">Checking that './plugins/translator/css/' is writable … warning</p>
    <p class="xh_warning">Checking that './plugins/translator/config/' is writable … warning</p>
  </section>
</article>
